src/pages/publication/contributors_tab/OtherContributor.tsx

```diff
@@ -1,72 +1,82 @@
 import ContributorType from '../../../types/contributor.types';
 import contributorTypes from '../../../utils/testfiles/contributor_types.json';
+import ContributorStyles from './StyledContributor';
 import StyledContributor from './StyledContributor';
+import { Field } from 'formik';
 
-const StyledNameInput = styled(TextField)`
+const StyledNameInput = styled(Field)`
   background-color: ${({ theme }) => theme.palette.background.default};
   margin-right: 1rem;
 `;
 
 interface OtherContributorProps {
   contributor: ContributorType;
-  dispatch: Dispatch<any>;
+  index: number;
+  setFieldValue: (name: string, value: any) => void;
+  swap: (indexA: number, indexB: number) => void;
+  remove: (index: number) => void;
 }
 
-const OtherContributor: React.FC<OtherContributorProps> = ({ contributor, dispatch }) => {
-  const handleTypeChange = (contributor: ContributorType) => (event: React.ChangeEvent<any>) => {
-    contributor.type = event.target.value;
-    dispatch(updateContributor(contributor));
-  };
-
-  const handleInstitutionChange = (contributor: ContributorType) => (event: React.ChangeEvent<any>) => {
-    contributor.selectedInstitution = event.target.value;
-    dispatch(updateContributor(contributor));
-  };
-
-  const handleNameChange = (contributor: ContributorType) => (event: React.ChangeEvent<any>) => {
-    contributor.name = event.target.value;
-    dispatch(updateContributor(contributor));
-  };
-
-  const deleteContributor = (contributor: ContributorType): void => {
-    dispatch(removeContributor(contributor));
-  };
-
+const OtherContributor: React.FC<OtherContributorProps> = ({ contributor, index, setFieldValue, swap, remove }) => {
   return (
-    <StyledContributor.OtherContributorContainer>
-      <StyledContributor.TypeSelect
-        onChange={handleTypeChange(contributor)}
-        value={contributor.type || ''}
-        variant="outlined">
-        <MenuItem value="" key="-1" />
-        {contributorTypes
-          .filter(type => type !== 'Author')
-          .map(type => (
-            <MenuItem value={type} key={type}>
-              {type}
-            </MenuItem>
-          ))}
-      </StyledContributor.TypeSelect>
-      <StyledNameInput value={contributor.name} onChange={handleNameChange(contributor)} variant="outlined" />
-      <StyledContributor.InstitutionSelect
-        onChange={handleInstitutionChange(contributor)}
-        value={contributor.selectedInstitution || ''}
-        variant="outlined">
-        <MenuItem value="" key="-1" />
-        {contributor?.institutions?.map(institution => (
-          <MenuItem value={institution} key={institution}>
-            {institution}
-          </MenuItem>
-        ))}
-      </StyledContributor.InstitutionSelect>
-      <StyledContributor.DeleteIcon onClick={() => deleteContributor(contributor)} />
-    </StyledContributor.OtherContributorContainer>
+    <Field name={`contributors.authors[${index}]`}>
+      {({ form: { values } }: any) => {
```

Comment:
`field: {values}`?

Reply:
eller nei, det går sikkert ikke siden field name er dynamisk..? Det gjelder kanskje andre plasser jeg har kommentert også i så fall

Reply:
Samme som over, trenger å vite størrelsen på arrayet som inneholder contributors for å rendre pilene